test/functional/feature_onion.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test bitcoind with 'onion'-tagged peers.
+
+Tested inbound and outbound peers.
+"""
+
+import os
+
+from test_framework.p2p import P2PInterface
+from test_framework.socks5 import Socks5Configuration, Socks5Server
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    PORT_MIN,
+    PORT_RANGE,
+    assert_equal,
+)
+
+
+class OnionTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+        self.setup_clean_chain = True
+
+    def setup_network(self):
+        self.add_nodes(self.num_nodes)
+        self.start_nodes()
+
+    def test_connections(self, in_clearnet=0, in_onion=0, out_onion=0):
+        self.restart_node(0, ['-bind={}=onion'.format(self.bind_onion), self.arg_onion])
+        self.restart_node(1)
```

Comment:
Stop-starting the nodes 11 times just to test new connections seems very inefficient. Is there a reason you need to do this stop-start?

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/20172#issuecomment-712763616).